examples/regression.py

```diff
@@ -0,0 +1,127 @@
+import argparse
+
+import functorch
+from functorch._src.compile_utils import strip_overloads
+import iree_torch
+import numpy as np
+from sklearn.datasets import make_regression
+from sklearn.model_selection import train_test_split
+import torch
+import torch_mlir
+
+
+# Dataset
+X, y, coef = make_regression(n_features=3, coef=True)
+X = torch.from_numpy(X).to(dtype=torch.float32)
+y = torch.from_numpy(y).to(dtype=torch.float32)
+X, X_test, y, y_test = train_test_split(X, y)
+
+# Weights
+w = torch.zeros(X.shape[1:])
+b = torch.tensor(0.)
+
+
+def _get_argparse():
+    parser = argparse.ArgumentParser(
+        description="Train and run a regression model.")
+    parser.add_argument("--iree-backend",
+        default="llvm-cpu",
+        help="See https://iree-org.github.io/iree/deployment-configurations/ "
+             "for the full list of options.")
+    return parser
+
+
+def _suppress_warnings():
+    import warnings
+    warnings.simplefilter("ignore")
+    import os
+
+
+def forward(w, b, X):
+    return torch.matmul(X, w) + b
+
+
+def mse(y_pred, y):
+    err = y_pred - y
+    return torch.mean(torch.square(err))
+
+
+def loss_fn(w, b, X, y):
+    y_pred = forward(w, b, X)
+    return mse(y_pred, y)
+
+
+grad_fn = functorch.grad(loss_fn, argnums=(0, 1))
+
+
+def update(w, b, grad_w, grad_b):
+    new_w = w - grad_w * 0.05
```

Comment:
Nit: put 0.05 into a variable "learning rate" for clarity

Reply:
Done.